koku/masu/api/upgrade_trino/util/task_handler.py

```diff
@@ -0,0 +1,134 @@
+from masu.celery.tasks import fix_parquet_data_types
+from masu.processor import is_customer_large
+from masu.processor.tasks import GET_REPORT_FILES_QUEUE
+from masu.processor.tasks import GET_REPORT_FILES_QUEUE_XL
+from masu.util.common import strip_characters_from_column_name
+from reporting.provider.aws.models import TRINO_REQUIRED_COLUMNS as AWS_TRINO_REQUIRED_COLUMNS
+from reporting.provider.azure.models import TRINO_REQUIRED_COLUMNS as AZURE_TRINO_REQUIRED_COLUMNS
+from reporting.provider.oci.models import TRINO_REQUIRED_COLUMNS as OCI_TRINO_REQUIRED_COLUMNS
+
+LOG = logging.getLogger(__name__)
+
+
+class RequiredParametersError(Exception):
+    """Handle require parameters error."""
+
+
+@dataclass
+class FixParquetTaskHandler:
+    start_date: Optional[str] = field(default=None)
+    provider_uuid: Optional[str] = field(default=None)
+    provider_type: Optional[str] = field(default=None)
+    simulate: Optional[bool] = field(default=False)
+    cleaned_column_mapping: Optional[dict] = field(default=None)
+
+    # Node role is the only column we add manually for OCP
+    # Therefore, it is the only column that can be incorrect
+    REQUIRED_COLUMNS_MAPPING = {
+        Provider.PROVIDER_OCI: OCI_TRINO_REQUIRED_COLUMNS,
+        Provider.PROVIDER_OCP: {"node_role": ""},
+        Provider.PROVIDER_AWS: AWS_TRINO_REQUIRED_COLUMNS,
+        Provider.PROVIDER_AZURE: AZURE_TRINO_REQUIRED_COLUMNS,
+    }
+
+    @classmethod
+    def from_query_params(cls, query_params: QueryDict) -> "FixParquetTaskHandler":
+        """Create an instance from query parameters."""
+        reprocess_kwargs = cls()
+        if start_date := query_params.get("start_date"):
+            if start_date:
+                reprocess_kwargs.start_date = parser.parse(start_date).replace(day=1)
+
+        if provider_uuid := query_params.get("provider_uuid"):
+            provider = Provider.objects.filter(uuid=provider_uuid).first()
+            if not provider:
+                raise RequiredParametersError(f"The provider_uuid {provider_uuid} does not exist.")
+            reprocess_kwargs.provider_uuid = provider_uuid
+            reprocess_kwargs.provider_type = provider.type
+
+        if provider_type := query_params.get("provider_type"):
+            reprocess_kwargs.provider_type = provider_type
+
+        if simulate := query_params.get("simulate"):
+            if simulate.lower() == "true":
+                reprocess_kwargs.simulate = True
+
+        if not reprocess_kwargs.provider_type and not reprocess_kwargs.provider_uuid:
+            raise RequiredParametersError("provider_uuid or provider_type must be supplied")
+        if not reprocess_kwargs.start_date:
+            raise RequiredParametersError("start_date must be supplied as a parameter.")
+
+        reprocess_kwargs.cleaned_column_mapping = reprocess_kwargs.clean_column_names(reprocess_kwargs.provider_type)
+        return reprocess_kwargs
```

Comment:
This is a very good use of the factory pattern. I would suggest changing the strategy used from creating a class instance and modifying it to building the parameters and passing those into the constructor. That way you can make the dataclass frozen, making the instance more durable.

Here is a rough example that probably has some room for further improvement.

```python
@dataclass(frozen=True)
class FixParquetTaskHandler:
    start_date: Optional[str] = field(default=None)
    provider_uuid: Optional[str] = field(default=None)
    provider_type: Optional[str] = field(default=None)
    simulate: Optional[bool] = field(default=False)
    cleaned_column_mapping: Optional[dict] = field(default=None)

    # Node role is the only column we add manually for OCP
    # Therefore, it is the only column that can be incorrect
    REQUIRED_COLUMNS_MAPPING = {
        Provider.PROVIDER_OCI: OCI_TRINO_REQUIRED_COLUMNS,
        Provider.PROVIDER_OCP: {"node_role": ""},
        Provider.PROVIDER_AWS: AWS_TRINO_REQUIRED_COLUMNS,
        Provider.PROVIDER_AZURE: AZURE_TRINO_REQUIRED_COLUMNS,
    }

    @classmethod
    def from_query_params(cls, query_params: QueryDict) -> "FixParquetTaskHandler":
        """Create an instance from query parameters."""
        kwargs = {}
        if start_date := query_params.get("start_date"):
            if start_date:
                kwargs["start_date"] = parser.parse(start_date).replace(day=1)

        if provider_uuid := query_params.get("provider_uuid"):
            provider = Provider.objects.filter(uuid=provider_uuid).first()
            if not provider:
                raise RequiredParametersError(f"The provider_uuid {provider_uuid} does not exist.")
            kwargs["provider_uuid"] = provider_uuid
            kwargs["provider_type"] = provider.type

        if provider_type := query_params.get("provider_type"):
            kwargs["provider_type"] = provider_type

        if simulate := query_params.get("simulate"):
            if simulate.lower() == "true":
                kwargs["simulate"] = True

        if not kwargs["provider_type"] and not kwargs["provider_uuid"]:
            raise RequiredParametersError("provider_uuid or provider_type must be supplied")
        if not kwargs["start_date"]:
            raise RequiredParametersError("start_date must be supplied as a parameter.")

        kwargs["cleaned_column_mapping"] = cls.clean_column_names(kwargs["provider_type"])
        
        return cls(**kwargs)
```

Reply:
Great suggestion @samdoran! Much cleaner approach here.